plugins/endpoints/src/main/java/org/pytorch/serve/plugins/endpoint/Token.java

```diff
@@ -0,0 +1,188 @@
+package org.pytorch.serve.plugins.endpoint;
+
+// import java.util.Properties;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.QueryStringDecoder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.SecureRandom;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.pytorch.serve.servingsdk.Context;
+import org.pytorch.serve.servingsdk.ModelServerEndpoint;
+import org.pytorch.serve.servingsdk.annotations.Endpoint;
+import org.pytorch.serve.servingsdk.annotations.helpers.EndpointTypes;
+import org.pytorch.serve.servingsdk.http.Request;
+import org.pytorch.serve.servingsdk.http.Response;
+
+@Endpoint(
+        urlPattern = "token",
+        endpointType = EndpointTypes.MANAGEMENT,
+        description = "Token authentication endpoint")
+public class Token extends ModelServerEndpoint {
+    private static String apiKey;
+    private static String managementKey;
+    private static String inferenceKey;
+    private static Instant managementExpirationTime;
+    private static Instant inferenceExpirationTime;
+    private static Integer timeToExpiration = 30;
+    private SecureRandom secureRandom = new SecureRandom();
+    private Base64.Encoder baseEncoder = Base64.getUrlEncoder();
+
+    @Override
+    public void doGet(Request req, Response rsp, Context ctx) throws IOException {
+        String queryResponse = parseQuery(req);
+        String test = "";
+        if ("management".equals(queryResponse)) {
+            generateKeyFile(1);
+        } else if ("inference".equals(queryResponse)) {
+            generateKeyFile(2);
+        } else {
+            test = "{\n\t\"Error\": " + queryResponse + "\n}\n";
+        }
+        rsp.getOutputStream().write(test.getBytes(StandardCharsets.UTF_8));
```

Comment:
Can we ensure https is enforced and send the new key through this secured channel? That way the caller would already poses the new key and can distribute to users.